src/domain.rs

```diff
@@ -30,16 +30,12 @@ impl SubscriberName {
         // Iterate over all characters in the input `s` to check if any of them matches
         // one of the characters in the forbidden array.
         let forbidden_characters = ['/', '(', ')', '"', '<', '>', '\\', '{', '}'];
-        let contains_forbidden_characters = s
-            .chars()
-            .filter(|g| forbidden_characters.contains(g))
-            .count()
-            > 0;
+        let contains_forbidden_characters = s.chars().any(|g| forbidden_characters.contains(&g));
 
         if is_empty_or_whitespace || is_too_long || contains_forbidden_characters {
             Err(format!("{} is not a valid subscriber name.", s))
         } else {
-            Ok(Self(s))
+            Ok(SubscriberName(s))
```

Comment:
I use `SubscriberName` here to make it easier to read and understand because the return is `SubscriberName`, if you want I can change it back to `Self`.

Reply:
I'd prefer `Self` here :bow:

Reply:
```suggestion
            Ok(Self(s))
```